Source/SwiftLintFramework/Rules/Lint/EmptyMethodOverrideRule.swift

```diff
@@ -0,0 +1,158 @@
+        }
+        """,
+        """
+        internal
+        class
+        override
+        func setUp() {
+            super.setUp()
+        }
+        """
+    ].map(wrapInClass)
+
+    static let corrections = Dictionary(uniqueKeysWithValues:
+        EmptyMethodOverrideRule.triggeringExamples.map { ($0, wrapInClass("")) })
+
+    public static let description = RuleDescription(
+        identifier: "empty_method_override",
+        name: "Empty Method Override",
+        description: "Methods that don't do anything but call `super` can be removed",
+        kind: .lint,
+        nonTriggeringExamples: EmptyMethodOverrideRule.nonTriggeringExamples,
+        triggeringExamples: EmptyMethodOverrideRule.triggeringExamples,
+        corrections: EmptyMethodOverrideRule.corrections
+    )
+
+    public func violationRanges(in file: File, kind: SwiftDeclarationKind,
+                                dictionary: [String: SourceKitRepresentable]) -> [NSRange] {
+        let overridingKinds: [SwiftDeclarationKind] = [
+            .functionMethodInstance,
+            .functionMethodClass,
+            .functionMethodStatic
+        ]
+        guard overridingKinds.contains(kind),
+            dictionary.enclosedSwiftAttributes.contains(.override),
+            !dictionary.enclosedSwiftAttributes.contains(.public),
+            file.onlyCallsSuper(dictionary),
+            let offset = dictionary.offset,
+            let length = dictionary.length
+            else { return [] }
+
+        let startIndex = dictionary.swiftAttributes
+            .compactMap { $0.offset }
+            .min() ?? offset
+        let endIndex = offset + length
+
+        return [NSRange(startIndex..<endIndex)]
+    }
+
+    public func validate(file: File, kind: SwiftDeclarationKind,
+                         dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+        return violationRanges(in: file, kind: kind, dictionary: dictionary).map {
+            StyleViolation(
+                ruleDescription: type(of: self).description,
+                severity: configuration.severity,
+                location: Location(file: file, byteOffset: $0.location)
+            )
+        }
+    }
+
+    public func substitution(for violationRange: NSRange, in file: File) -> (NSRange, String) {
+        let range = file.contents.bridge()
+            .byteRangeToNSRange(start: violationRange.location, length: violationRange.length)!
+        return (range, "")
+    }
+}
+
+private extension File {
+    func onlyCallsSuper(_ dictionary: [String: SourceKitRepresentable]) -> Bool {
+        if let name = dictionary.name?.split(separator: "(").first,
+            dictionary.substructure.count == 1,
+            let methodCall = dictionary.substructure.first,
+            methodCall.name == "super.\(name)",
+            !hasAssignmentInBody(dictionary) {
+            return true
+        } else {
+            return false
+        }
```

Comment:
Here, you're doing it nearly as I suggest, but having the beginning of the condition on the same line like `if` doesn't give a nice balance, also it's not clear where the body of the `if` begins, so I'd rather do this:

```swift
        if 
            let name = dictionary.name?.split(separator: "(").first,
            dictionary.substructure.count == 1,
            let methodCall = dictionary.substructure.first,
            methodCall.name == "super.\(name)",
            !hasAssignmentInBody(dictionary)
        {
            return true
        } else {
            return false
        }
```

Reply:
I agree about the balance, but wasn't sure about Allman in Swift, so I didn't go for it at first. Looks better this way, so let's do it like this.